### PR TITLE
Use const refs to reduce copying.

### PR DESCRIPTION
**Pull request by waywardmonkeys**
These are things that have been found by `clang-tidy`.

### CODE REVIEW

**Comment by waywardmonkeys** [comment]
Is there anything that I can do to help this land sooner? I have other follow up patches to submit after this one ...

File: src/muz/rel/dl_instruction.cpp [hunk 1]
```diff
@@ -191,7 +191,7 @@ namespace datalog {
         reg_idx m_reg;
     public:
         instr_io(bool store, func_decl_ref pred, reg_idx reg)
```
Comment (NikolajBjorner) [inline src/muz/rel/dl_instruction.cpp]: TBD make it const&

File: src/ast/proofs/proof_utils.h [hunk 2]
```diff
@@ -88,7 +88,7 @@ class elim_aux_assertions {
 
     app_ref m_aux;
 public:
-    elim_aux_assertions(app_ref aux) : m_aux(aux) {}
+    elim_aux_assertions(app_ref aux) : m_aux(std::move(aux)) {}
```
Comment (NikolajBjorner) [inline src/ast/proofs/proof_utils.h]: TBD make it const&

File: src/muz/rel/dl_product_relation.cpp [hunk 3]
```diff
@@ -488,7 +488,7 @@ namespace datalog {
         ptr_vector<relation_transformer_fn> m_transforms;
     public:
         transform_fn(relation_signature s, unsigned num_trans, relation_transformer_fn** trans):
```
Comment (NikolajBjorner) [inline src/muz/rel/dl_product_relation.cpp]: TBD make it const&

**Comment by NikolajBjorner** [comment]
It very good. Thanks

**Pull request merged**